sktime/forecasting/online_ensemble.py

```diff
@@ -0,0 +1,171 @@
+
+    def update_predict(self, y_test, X_test=None, update_params=False,
+                       return_pred_int=False,
+                       alpha=DEFAULT_ALPHA):
+        """Make and update predictions iteratively over the test set.
+
+        Parameters
+        ----------
+        y_test : pd.Series
+        cv : temporal cross-validation generator, optional (default=None)
+        X_test : pd.DataFrame, optional (default=None)
+        update_params : bool, optional (default=False)
+        return_pred_int : bool, optional (default=False)
+        alpha : int or list of ints, optional (default=None)
+
+        Returns
+        -------
+        y_pred : pd.Series
+            Point predictions
+        y_pred_int : pd.DataFrame
+            Prediction intervals
+        """
+
+        return self._predict_moving_cutoff(y_test, X=X_test,
+                                           update_params=update_params,
+                                           return_pred_int=return_pred_int,
+                                           alpha=alpha)
+
+    def _predict_moving_cutoff(self, y, cv=OnlineSplitter(), X=None,
+                               update_params=False,
+                               return_pred_int=False,
+                               alpha=DEFAULT_ALPHA):
+        """Make single-step or multi-step moving cutoff predictions
+
+        Parameters
+        ----------
+        y : pd.Series
+        cv : temporal cross-validation generator, set to OnlineSplitter()
+        X : pd.DataFrame
+        update_params : bool
+        return_pred_int : bool
+        alpha : float or array-like
+
+        Returns
+        -------
+        y_pred = pd.Series
+        """
+        if return_pred_int:
+            raise NotImplementedError()
+        fh = np.array([1])
+        y_preds = []
+        cutoffs = []
+        with self._detached_cutoff():
+            self._set_cutoff(y.index[0] - 1)
+            for new_window, _ in cv.split(y):
+                y_new = y.iloc[new_window]
+
+                # we cannot use update_predict_single here, as this would
+                # re-set the forecasting horizon, instead we use
+                # the internal _update_predict_single method
+                y_pred = self._update_predict_single(
+                    y_new, fh, X=X,
+                    update_params=update_params,
+                    return_pred_int=return_pred_int,
+                    alpha=alpha)
+                y_preds.append(y_pred)
+                cutoffs.append(self.cutoff)
+        return _format_moving_cutoff_predictions(y_preds, cutoffs)
+
+    def _predict(self, fh, X=None, return_pred_int=False, alpha=DEFAULT_ALPHA):
+        if return_pred_int:
+            raise NotImplementedError()
+
+        return (pd.concat(
+            self._predict_forecasters(fh=fh, X=X), axis=1)
+                * self.ensemble_algorithm.weights).sum(axis=1)
+
+
+def _format_moving_cutoff_predictions(y_preds, cutoffs):
```

Comment:
Have you changed anything in this function? Otherwise we could import it

Reply:
Importing it instead